jni/themis_jni.mk

```diff
@@ -41,7 +41,45 @@ $(BIN_PATH)/$(LIBTHEMISJNI_SO): $(THEMIS_JNI_OBJ) $(THEMIS_STATIC)
 	@mkdir -p $(@D)
 	@echo -n "link "
 	@$(BUILD_CMD)
+
+themis_jni_install: $(BIN_PATH)/$(LIBTHEMISJNI_SO)
+	@echo -n "install Themis JNI "
+	@mkdir -p $(DESTDIR)$(jnidir)
+	@$(INSTALL_PROGRAM) $(BIN_PATH)/$(LIBTHEMISJNI_SO) $(DESTDIR)$(jnidir)
 ifdef IS_MACOS
-	@install_name_tool -id "$(PREFIX)/lib/$(notdir $@)" $(BIN_PATH)/$(notdir $@)
-	@install_name_tool -change "$(BIN_PATH)/$(notdir $@)" "$(PREFIX)/lib/$(notdir $@)" $(BIN_PATH)/$(notdir $@)
+	@install_name_tool -id "$(jnidir)/$(LIBTHEMISJNI_SO)" "$(DESTDIR)$(jnidir)/$(LIBTHEMISJNI_SO)"
+	@install_name_tool -change "$(BIN_PATH)/$(LIBTHEMIS_SO)" "$(libdir)/$(LIBTHEMIS_SO)" "$(DESTDIR)$(jnidir)/$(LIBTHEMISJNI_SO)"
 endif
+	@$(PRINT_OK_)
+	@java_library_path=$$(\
+	    java -XshowSettings:properties -version 2>&1 \
+	    | sed -E 's/^ +[^=]+ =/_&/' \
```

Comment:
black magic happening here

Reply:
Well, if you'd like it can be replaced with the following Java code:

```java
import java.io.File;

public final class Main {
    public static void main(String[] args) {
        String rawPath = System.getProperty("java.library.path");
        String[] paths = rawPath.split(File.pathSeparator);
        for (String path : paths) {
            System.out.println(path);
        }
    }
}
```

Just need to compile and run it. Or we can bundle a precompiled *.jar.